app/controllers/crypto_address_for_channels_controller.rb

```diff
@@ -0,0 +1,114 @@
+require "eth"
+require "rbnacl"
+require "base58"
+
+class CryptoAddressForChannelsController < ApplicationController
+  include Eth
+  before_action :authenticate_publisher!
+
+  def index
+    current_channel = current_publisher.channels.find(params[:channel_id])
+    @crypto_addresses_for_channel = CryptoAddressForChannel.where(channel: current_channel)
+    render(json: @crypto_addresses_for_channel)
+  end
+
+  def generate_nonce
+    set = [('a'..'z'), ('A'..'Z'), (0..9)].map(&:to_a).flatten
+    nonce = (0...32).map { set[rand(set.length)] }.join
```

Comment:
Abusing the random number generation here is very high complexity, but still whenever we are generating random numbers for security purposes we should use a cryptographically secure random number generator. @kdenhartog do we want to recommend [Ruby's securerandom](https://ruby-doc.org/stdlib-2.4.0/libdoc/securerandom/rdoc/SecureRandom.html)? e.g. `SecureRandom.hex(16)`.

Reply:
The Eth library I'm using for ruby on the back end seems to do something to hex strings internally, and will always return false on verification when I use `SecureRandom`.  I talked about this with @kdenhartog, and I think this is random enough for our purposes.  The ruby libraries have just been a nightmare.

Reply:
When I was looking at that, I was thinking we'd do something like `nonce = Base58Encode(SecureRandom(16))`. This is just pseudocode and I'm not sure of the specific syntax to achieve this in Ruby off the top of my head. 

Ultimately, we do want to make sure we're using something sufficiently random with a CSPRG like SecureRandom still (at least 128 bits of entropy is ideal), but it's fine to use a different encoding other than hex encoding.

Reply:
Looks like `SecureRandom.uuid` would be an ideal option here if you want to remove any sort of encoding confusion.